utils/vehnt/src/cli/delegated.rs

```diff
@@ -0,0 +1,399 @@
+  let memcmp = RpcFilterType::Memcmp(Memcmp::new_base58_encoded(0, &input));
+  config.filters = Some(vec![RpcFilterType::DataSize(196), memcmp]);
+  config.account_config.encoding = Some(UiAccountEncoding::Base64);
+  let accounts = rpc_client
+    .get_program_accounts_with_config(&helium_dao_id, config)
+    .await?;
+  Ok(accounts)
+}
+
+pub struct FullPosition {
+  pub position_key: Pubkey,
+  pub delegated_position_key: Pubkey,
+  pub position: PositionV0,
+  pub delegated_position: DelegatedPositionV0,
+}
+
+const BATCH_SIZE: usize = 100;
+impl Delegated {
+  pub async fn run(self, rpc_client: RpcClient, solana_url: String) -> MyResult {
+    let mut total_hnt = 0_u64;
+    let mut total_vehnt = 0_u128;
+    let mut mobile_vehnt = 0_u128;
+    let mut iot_vehnt = 0_u128;
+    let mut iot_fall_rate = 0_u128;
+    let mut mobile_fall_rate = 0_u128;
+
+    let mut curr_ts = self.curr_ts;
+
+    println!("Current ts: {}", curr_ts);
+    let accounts = get_stake_accounts(&rpc_client).await?;
+    let delegated_positions = accounts
+      .iter()
+      .map(|(pubkey, account)| {
+        let mut data = account.data.as_slice();
+        DelegatedPositionV0::try_deserialize(&mut data).map(|position| (pubkey, position))
+      })
+      .map(|result| result.unwrap())
+      .collect::<Vec<_>>();
+
+    let position_keys = delegated_positions
+      .iter()
+      .map(|(_pubkey, delegated_position)| delegated_position.position)
+      .collect::<Vec<Pubkey>>();
+
+    let mut positions_raw = Vec::new();
+    for chunk in position_keys.chunks(BATCH_SIZE) {
+      let chunk_positions_raw = rpc_client.get_multiple_accounts(chunk).await?;
+      positions_raw.extend(chunk_positions_raw);
+    }
+
+    let positions_with_delegations: Vec<FullPosition> = delegated_positions
+      .iter()
+      .zip(positions_raw)
+      .map(|((pubkey, delegated_position), position)| {
+        let position_unwrapped = position.unwrap();
+        let mut data = position_unwrapped.data.as_slice();
+        let position_parsed = PositionV0::try_deserialize(&mut data).unwrap();
+        FullPosition {
+          position_key: delegated_position.position,
+          delegated_position_key: **pubkey,
+          position: position_parsed,
+          delegated_position: delegated_position.clone(),
+        }
+      })
+      .collect();
+
+    let registrar_keys: Vec<Pubkey> = positions_with_delegations
+      .iter()
+      .map(|p| p.position.registrar)
+      .collect::<HashSet<_>>()
+      .into_iter()
+      .collect();
+    let registrars_raw = rpc_client.get_multiple_accounts(&registrar_keys).await?;
+    let registrars: Vec<Registrar> = registrars_raw
+      .iter()
+      .map(|registrar| {
+        let mut data = registrar.as_ref().unwrap().data.as_slice();
+        Registrar::try_deserialize(&mut data)
+      })
+      .map(|result| result.unwrap())
+      .collect();
+
+    let voting_mint_config = registrars[0].voting_mints[0].clone();
```

Comment:
Add comment stating that this could be incorrect in the future as registrars are added